integ/lib/ThinkboxDockerImageOverrides.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Repository } from '@aws-cdk/aws-ecr';
+import { ContainerImage } from '@aws-cdk/aws-ecs';
+import { Construct } from '@aws-cdk/core';
+import {
+  RenderQueueImages,
+  ThinkboxManagedDeadlineDockerRecipes as Recipes,
+  UsageBasedLicensingImages,
+} from 'aws-rfdk/deadline';
+
+type DeadlineDockerImageOverrides = {[key in Recipes]?: string};
+
+/**
+ * Interface for Docker image overrides stored in an ECR repository.
+ */
+interface IECRImageOverrides {
+  /**
+   * The ARN of the ECR repository.
+   */
+  readonly repositoryArn: string;
+  /**
+   * A mapping of Deadline Docker recipe names (see ThinkboxManagedDeadlineDockerRecipes in aws-rfdk) to tags of the override Docker images
+   */
+  readonly imageOverrides: DeadlineDockerImageOverrides;
+}
+
+/**
+ * Implementation of {@link IECRImageOverrides}.
+ */
+class ECRImageOverrides implements IECRImageOverrides {
+  /**
+   * Creates an {@link ECRImageOverrides} from a JSON string.
+   * @param json The JSON string.
+   */
+  public static fromJSON(json: string): ECRImageOverrides {
+    const obj = JSON.parse(json);
+
+    // Validate the input JSON
+    const errors = [];
+    [
+      'repositoryArn',
+      'imageOverrides',
+    ].forEach(prop => {
+      if (!(prop in obj)) {
+        errors.push(`Property ${prop} was expected but not found in ${obj}`);
+      }
+    });
+    if (!(obj.imageOverrides instanceof Object)) {
+      errors.push(`Expected Object for imageOverrides but found ${typeof(obj.imageOverrides)}`);
+    }
+    Object.keys(obj.imageOverrides).forEach(key => {
+      if (!Object.values(Recipes).includes(key as Recipes)) {
+        errors.push(`Key ${key} in imageOverrides is invalid. Must be in: [${Object.values(Recipes)}].`);
+      }
+    });
+
+    if (errors.length > 0) {
+      throw new Error(`Invalid JSON for ECRImageOverrides: ${errors.join('\n')}`);
+    }
+
+    return new ECRImageOverrides({
```

Comment:
nit: Can we simplify it to:
`return obj as ECRImageOverrides;`

Reply:
Sure, updated